trieste/bayesian_optimizer.py

```diff
@@ -39,47 +40,65 @@
 
 
 @dataclass(frozen=True)
-class LoggingState(Generic[S]):
-    """
-    Container for the state of the optimization process in :class:`BayesianOptimizer` at each
-    optimization step.
-    """
+class Record(Generic[S]):
+    """ Container to record the state of each step of the optimization process. """
 
     datasets: Mapping[str, Dataset]
-    """ All observer data at this optimization step. """
+    """ The known data from the observer. """
 
     models: Mapping[str, TrainableProbabilisticModel]
     """ The models over the :attr:`datasets`. """
 
     acquisition_state: Optional[S]
-    """ The acquisition state after this optimization step. """
+    """ The acquisition state. """
 
 
+# this should be a generic NamedTuple, but mypy doesn't support them
+#  https://github.com/python/mypy/issues/685
 @dataclass(frozen=True)
 class OptimizationResult(Generic[S]):
-    """ Container for the final result of the optimization process. """
+    """ The final result, and the historical data of the optimization process. """
 
-    datasets: Mapping[str, Dataset]
+    final_result: Result[Record[S]]
     """
-    All data from the observer (unless :attr:`error` is populated, in which case this is the data
-    from the point at which the process was interrupted).
+    The final result of the optimization process. This contains either a :class:`Record` or an
+    exception.
     """
 
-    models: Mapping[str, TrainableProbabilisticModel]
-    """
-    The models over the :attr:`datasets` (unless :attr:`error` is populated, in which case this is
-    the models from the point at which the process was interrupted). """
+    history: List[Record[S]]
+    r""" The history of the :class:`Record`\ s from each step of the optimization process. """
```

Comment:
Could you clarify whether history[-1] == final_result?

Reply:
will do